internal/experiment/generation/transformer.go

```diff
@@ -262,21 +262,36 @@ func parameterNamer(selected []interface{}) ParameterNamer {
 
 	return func(meta yaml.ResourceMeta, path []string, name string) string {
 		var parts []string
+
 		if needsKind {
-			parts = append(parts, strings.ToLower(meta.Kind))
+			parts = append(parts, meta.Kind)
 		}
+
 		if needsName {
-			parts = append(parts, strings.Split(meta.Name, "-")...)
+			parts = append(parts, meta.Name)
 		}
+
 		if needsPath[meta.Kind][meta.Name] > 1 {
 			for _, p := range path {
 				if yaml.IsListIndex(p) {
 					if _, value, _ := yaml.SplitIndexNameValue(p); value != "" {
-						parts = append(parts, strings.Split(value, "-")...)
+						parts = append(parts, value)
 					}
 				}
 			}
 		}
-		return strings.Join(append(parts, name), "_")
+
+		if name != "" {
+			parts = append(parts, name)
+		}
+
+		// Explainer: Parameter names are used in Go Templates which are executed
+		// against Go structs, if the template parser encounters a token that is
+		// not a valid Go field name, parsing fails (e.g. "bad character U+002D '-'").
+
+		parameterName := strings.Join(parts, "_")
```

Comment:
do we need to worry about parameterName length <=63 chars ?

Reply:
The parameter name only lives in our API space, so it shouldn't need to be limited at all (it is just persisted on the backend as JSON). The only reason we need to worry about the "_" is the Go text/template parser and I think we can get around that restriction in our own template Engine by passing maps instead of structs as the `data` to the text/template `Execute` function (which is just kind of a pain). I ran some quick tests and something like `{{.Values.foo-bar}}` works fine if `data` is a `map[string]interface{}` but fails if `data` is a `struct` (and I suspect that `data` cannot be a `struct` with a `map` field, since that is our current implementation in the template Engine).